src/Build/Logging/TerminalLogger/TerminalLogger.cs

```diff
@@ -251,30 +251,56 @@ internal TerminalLogger(LoggerVerbosity verbosity, uint? originalConsoleMode) :
     public static ILogger CreateTerminalOrConsoleLogger(string[]? args = null)
     {
         (bool supportsAnsi, bool outputIsScreen, uint? originalConsoleMode) = NativeMethodsShared.QueryIsScreenAndTryEnableAnsiColorCodes();
+        var (logger, _) = CreateTerminalOrConsoleLoggerWithForwarding(args, supportsAnsi, outputIsScreen, originalConsoleMode);
+        return logger;
+    }
 
-        return CreateTerminalOrConsoleLogger(args, supportsAnsi, outputIsScreen, originalConsoleMode);
+    /// <summary>
+    /// Creates a Terminal logger if possible, or a Console logger. If the created logger supports remote logging,
+    /// also provides a ForwardingLoggerRecord to wrap it for forwarding.
+    /// </summary>
+    /// <param name="args">Command line arguments for the logger configuration. Currently, only 'tl|terminallogger', 'v|verbosity', 'tlp|terminalloggerparameters', and 'clp|consoleloggerparameters' are supported.</param>
+    public static (ILogger, ForwardingLoggerRecord?) CreateTerminalOrConsoleLoggerWithForwarding(string[]? args = null)
+    {
+        (bool supportsAnsi, bool outputIsScreen, uint? originalConsoleMode) = NativeMethodsShared.QueryIsScreenAndTryEnableAnsiColorCodes();
+        var (logger, forwardingLogger) = CreateTerminalOrConsoleLoggerWithForwarding(args, supportsAnsi, outputIsScreen, originalConsoleMode);
+        return (logger, forwardingLogger);
     }
 
-    internal static ILogger CreateTerminalOrConsoleLogger(string[]? args, bool supportsAnsi, bool outputIsScreen, uint? originalConsoleMode)
+    internal static (ILogger, ForwardingLoggerRecord?) CreateTerminalOrConsoleLoggerWithForwarding(string[]? args, bool supportsAnsi, bool outputIsScreen, uint? originalConsoleMode)
     {
         LoggerVerbosity verbosity = LoggerVerbosity.Normal;
         string tlEnvVariable = Environment.GetEnvironmentVariable("MSBUILDTERMINALLOGGER") ?? string.Empty;
-        string tlArg = string.Empty;
-        string? verbosityArg = string.Empty;
+        string? tlArg = null;
+        string? tlpArg = null;
+        string? clpArg = null;
+        string? verbosityArg = null;
+
+        ILogger loggerToReturn;
+        ForwardingLoggerRecord? forwardingLogger;
+
+        Regex tlArgRegex = new(@"(?:/|-|--)(?:tl|terminallogger):(?'value'on|off|true|false|auto)", RegexOptions.IgnoreCase);
+        Regex verbosityArgRegex = new(@"(?:/|-|--)(?:v|verbosity):(?'value'\w+)", RegexOptions.IgnoreCase);
+        Regex tlpArgRegex = new(@"(?:/|-|--)(?:tlp|terminalloggerparameters):(?'value'.+)", RegexOptions.IgnoreCase);
+        Regex clpArgRegex = new(@"(?:/|-|--)(?:clp|consoleloggerparameters):(?'value'.+)", RegexOptions.IgnoreCase);
 
         if (args != null)
         {
-            string argsString = string.Join(" ", args);
+            foreach (var arg in args)
```

Comment:
looks wrong, wouldn't this overwrite already parsed arguments?
if we pass 
["-tl:off", "-v:diag"]
it'll have 2 iterations, on first set the tlArg to "off" and then on second iteration it'll undo it and set verbostity

should be gated to assign only if matched